INFX574 lab 2

Name:Neha Palsokar

collaborators: Sayali Chakradeo, Harkar Singh   

In [2]:
import numpy as np
import pandas as pd
from sklearn.datasets import load_boston

import sklearn.linear_model as lm
import statsmodels.formula.api as smf

import matplotlib.pyplot as plt
%matplotlib inline

In [13]:
#Q1
titanic = pd.read_csv('titanic.csv.bz2')

In [14]:
titanic.head()

pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

       age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.0000      0      0   24160  211.3375       B5        S    2    NaN   
1   0.9167      1      2  113781  151.5500  C22 C26        S   11    NaN   
2   2.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.0000      1      2  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.0000      1      2  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [56]:
#Q2
mod=smf.ols(formula='survived ~ pclass + sex',data=titanic)
res = mod.fit()
print(res.summary())

                            OLS Regression Results                            
Dep. Variable:               survived   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.340
Method:                 Least Squares   F-statistic:                     338.3
Date:                Wed, 11 Apr 2018   Prob (F-statistic):          4.09e-119
Time:                        22:46:25   Log-Likelihood:                -639.31
No. Observations:                1309   AIC:                             1285.
Df Residuals:                    1306   BIC:                             1300.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept       1.0405      0.034     30.888      

In [57]:
#Q3: Predict the survival:
yhat=res.predict()
print(min(yhat))
print(max(yhat))

0.09980047253427904
0.8952274211186283


In [49]:
#Q4
mod2=smf.ols(formula='survived ~ C(pclass) + sex',data=titanic)
res2 = mod2.fit()
print(res2.summary())

yhat2=res2.predict()
#min and max values:
print(min(yhat2))
print(max(yhat2))
#values are not equal to 0 or 1

                            OLS Regression Results                            
Dep. Variable:               survived   R-squared:                       0.341
Model:                            OLS   Adj. R-squared:                  0.340
Method:                 Least Squares   F-statistic:                     225.5
Date:                Wed, 11 Apr 2018   Prob (F-statistic):          7.79e-118
Time:                        22:02:24   Log-Likelihood:                -639.22
No. Observations:                1309   AIC:                             1286.
Df Residuals:                    1305   BIC:                             1307.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept          0.8990      0.025     35.

In [53]:
#Q5 and Q6:

survived=titanic[(titanic.survived==1)]
actual_survival_rate=len(survived)/len(titanic)
print(actual_survival_rate)
#predicted survival rate:
print(np.mean(yhat>0.5))
#accuracy
print(np.mean(titanic.survived==(yhat>0.5)))

0.3819709702062643
0.3559969442322384
0.7799847211611918


In [54]:
#Q7
mod1=smf.ols(formula='survived ~ C(pclass)*sex',data=titanic)
res1 = mod1.fit()
print(res1.summary())

                            OLS Regression Results                            
Dep. Variable:               survived   R-squared:                       0.370
Model:                            OLS   Adj. R-squared:                  0.368
Method:                 Least Squares   F-statistic:                     153.2
Date:                Wed, 11 Apr 2018   Prob (F-statistic):          4.02e-128
Time:                        22:45:16   Log-Likelihood:                -609.86
No. Observations:                1309   AIC:                             1232.
Df Residuals:                    1303   BIC:                             1263.
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                                 coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------
Intercept           

#Q8

Suvived Male and class 3 : Intercept + C(sex)[T.male] + C(pclass)[T.3] + C(pclass)[T.3]:C(sex)[T.male]
                           =0.9653 -0.6245 -0.4745 +0.2859=0.1522
                 
Suvived Female and class 3: Intercept + C(pclass)[T.3]
                            =0.9653-0.4745=0.4908
                 
Suvived Female and class 1:Intercept
                           =0.9653

In [55]:
#Q9
yhat1=res1.predict()
print(min(yhat1))
print(max(yhat1))

0.14619883040935766
0.9652777777777782


In [52]:
#predicted survival rate:
print(np.mean(yhat1>0.5))
#accuracy
print(np.mean(titanic.survived==(yhat1>0.5)))

0.19098548510313215
0.7830404889228418


#Q9
Old accuracy :0.7799847211611918
New accuracy :0.7830404889228418
    
Improvement in accuracy is 0.003 which shows that the effect of gender variable is very less
